matRad/optimization/matRad_DoseOptimizationFunction.m

```diff
@@ -56,17 +56,58 @@
     
     % Helper methods
     methods (Access = public)
-        function doseParams = getDoseParameters(obj)
+        function objParams = getDoseParameters(obj)
             % get only the dose related parameters.
-            ix = cellfun(@(c) isequal('dose',c),obj.parameterTypes);
-            doseParams = [obj.parameters{ix}];
+            if isempty(obj.parameterTypes)
+                objParams = [];
+            end
+
+            if cellfun(@(c) isequal('dose',c),obj.parameterTypes)
+                ix = cellfun(@(c) isequal('dose',c),obj.parameterTypes);
+                objParams = [obj.parameters{ix}];
+            end
+
+            if cellfun(@(c) isequal('dirtyDose',c),obj.parameterTypes)
+                ix = cellfun(@(c) isequal('dirtyDose',c),obj.parameterTypes);
+                objParams = [obj.parameters{ix}];
+            end
+
+            if cellfun(@(c) isequal('LETxd',c),obj.parameterTypes)
+                ix = cellfun(@(c) isequal('LETxd',c),obj.parameterTypes);
+                objParams = [obj.parameters{ix}];
+            end
+
+            if cellfun(@(c) isequal('LETd',c),obj.parameterTypes)
```

Comment:
LET should not be in here. This function is used to adapt the prescription in according to fractionation. LET is independent of that.